### ISSUE TITLE
Add docstrings to methods

### ISSUE THREAD

**Issue by IvoMerchiers** [fabiocaccamo/python-benedict#2]
When programming using the python-benedict, I like to use the fact that my IDE can automatically show docstrings for methods. This way, you don't have to constantly look up the source code or guess the meaning of parameters. 

You pretty much have all of this information in the readme.md, so it might be good to copy that into your code.
Ideally, you can then automatically generate more extensive and searchable documentation with sphinx.

---
Example:

```
>>> from benedict import benedict
>>> test_dict = benedict({"a": 1})
>>> help(test_dict.clean)

Help on method clean in module benedict.dicts:

clean(strings=True, dicts=True, lists=True) method of benedict.dicts.benedict instance
```
This is not informative and should return a proper docstring.

**Comment by fabiocaccamo** [comment]
@IvoMerchiers thank you for reporting this, I will add docstrings as soon as possible.

**Comment by fabiocaccamo** [comment]
@IvoMerchiers released in `0.14.0`.
